src/runtime/scheduler.go

```diff
@@ -253,8 +261,13 @@ func scheduler() {
 				// It would be nice if we could detect deadlocks here, because
 				// there might still be functions waiting on each other in a
 				// deadlock.
+				// Due to interrupts, we may actually not be done - so we need to sleep.
 				scheduleLog("  no tasks left!")
-				return
+				sleepTicks(0)
+				if asyncScheduler {
+					break
+				}
```

Comment:
I _think_ this should be moved before the `sleepTicks` call. Sleeping in WebAssembly works a bit differently and I can only imagine sleeping _inside the scheduler_ will lead to bugs.

(Also, calls into the WebAssembly module from JavaScript could also be considered interrupts so wasm isn't free of them).

Reply:
Ah yeah I see now how this might work a bit differently on wasm.